PKIndex/PKIndex/PKIndexTableViewController.m

```diff
@@ -0,0 +1,295 @@
+//
+//  PKIndexTableViewController.m
+//  PKIndex
+//
+//  Created by Fatima Zenine Villanueva on 7/27/15.
+//  Copyright (c) 2015 apps. All rights reserved.
+//
+
+#import "PKIndexTableViewController.h"
+#import "PKIndexList.h"
+#import "PKDetailViewController.h"
+#import "AFHTTPRequestOperation.h"
+#import "AFNetworking.h"
+
+
+@interface PKIndexTableViewController ()
+@property (nonatomic) NSDictionary *pokemonIndex;
+@property (strong, nonatomic) IBOutlet UITableView *table;
+@property (strong, nonatomic) IBOutlet UISegmentedControl *clickMenu;
+@property(nonatomic,readonly) CGSize size;
+@end
+
+@implementation PKIndexTableViewController
+
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.normalType = [[NSMutableArray alloc]init];
+    self.fightingType = [[NSMutableArray alloc]init];
+    self.flyingType = [[NSMutableArray alloc]init];
+    self.poisonType = [[NSMutableArray alloc]init];
+    self.groundType = [[NSMutableArray alloc]init];
+    self.rockType = [[NSMutableArray alloc]init];
+    self.bugType = [[NSMutableArray alloc]init];
+    self.ghostType = [[NSMutableArray alloc]init];
+    self.steelType = [[NSMutableArray alloc]init];
+    self.fireType = [[NSMutableArray alloc]init];
+    self.waterType = [[NSMutableArray alloc]init];
+    self.grassType = [[NSMutableArray alloc]init];
+    self.electricType = [[NSMutableArray alloc]init];
+    self.psychicType = [[NSMutableArray alloc]init];
+    self.iceType = [[NSMutableArray alloc]init];
+    self.dragonType = [[NSMutableArray alloc]init];
+    self.fairyType = [[NSMutableArray alloc]init];
+    
+    self.pokemonIndex = @{ @"Pokemon":[PKIndexList allPokemon] };
+    
+    
+    NSIndexPath *indexPath;
+    NSArray *keys = [self.pokemonIndex allKeys];
+    NSString *key = keys[indexPath.section];
+    NSArray *pokemonArray = [self.pokemonIndex objectForKey:key];
+    NSLog(@"%@", pokemonArray);
+    
+    for (int i = 0; i < [pokemonArray count]; i++) {
+        
+        int idx = i;
+        NSString *apiURL = [NSString stringWithFormat:@"http://pokeapi.co/api/v1/pokemon/%u", idx + 1];
```

Comment:
This can create performance issues, you're basically making a separate web request for each and every pokemon.  This is a lot of web requests and back and forth, and isn't really scalable.

A good API will often provide ways to search and filter data so you have to do very few requests.  For example, get pokemon by type.  I actually couldn't find this in the Pokemon API unfortunately!